packages/world/ts/actions/delegation.ts

```diff
@@ -0,0 +1,56 @@
+import type { WalletClient, Transport, Chain, Account, Hex, WalletActions, WriteContractReturnType } from "viem";
+import { getAction, getAbiItem, formatAbiItem, toFunctionSelector, type GetAbiItemParameters } from "viem/utils";
+import { writeContract } from "viem/actions";
+import { encodeSystemCallFrom, type SystemCallFrom } from "../encodeSystemCallFrom";
+
+type DelegationParameters = {
+  worldAddress: Hex;
+  delegatorAddress: Hex;
+  getSystemId: (functionSelector: Hex) => Hex;
```

Comment:
```suggestion
  getSystemId: (functionSelector: Hex) => Promise<Hex>;
```

We should probably make this async because in the future 1) our client query APIs will be async and 2) we may wanna automatically retrieve this from the world